### PR TITLE
Feature/awx jobs status

### LINKED ISSUE #9: Get job templates last statuses from AWX

### PR DESCRIPTION
**Pull request by Polina-Gubina**
Implement bot `/status` command handling, i.e. `/status awx` retrieving status of AWX templates last job status

Resolve #9 (does not implement history size configuration)

### CODE REVIEW

File: delatore/sources/awx_api.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        json_data = self.get_templates(_filter)
+        list_templates_statuses = get_templates_statuses_from_json(json_data)
+        message = '\n\n'.join(str(stt) for stt in list_templates_statuses)
+        return message
+
+    def get_templates(self, filters: dict = None):
+        """Returns list of all job templates for csm organization
+
+        This methods support ansible tower filtering https://docs.ansible.com/ansible-tower/latest/html/towerapi/filtering.html
+        Supported filters:
```
Comment (outcatcher) [inline delatore/sources/awx_api.py]: Better remove this row

File: delatore/sources/awx_api.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            return response_data['results']
+        except KeyError:
+            LOGGER.error('No `results` field found in /job_templates response: \nResponse: %s', response_data)
+            raise
+
+    def get_api_endpoints(self):
+        """
+        Returns list of all awx api endpoints
+
+        :return:
```
Comment (outcatcher) [inline delatore/sources/awx_api.py]: Can be single-line doc

File: delatore/sources/awx_api.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    """Get status for all templates or for concrete template"""
+    list_template_status = []
+    for i in json_data:
+        list_template_status.append(
+            TemplateStatus(name=i['name'], last_run_timestamp=i['last_job_run'], last_status=i['status'],
+                           playbook=i['playbook']))
+    return list_template_status
+
+
+'''
```
Comment (outcatcher) [inline delatore/sources/awx_api.py]: Should be removed 🙈

File: delatore/sources/awx_api.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        :return:
+        """
+        response = self.session.get(url=self.url)
+        return response.json()
+
+
+def get_templates_statuses_from_json(json_data) -> List[TemplateStatus]:
+    """Get status for all templates or for concrete template"""
+    list_template_status = []
+    for i in json_data:
```
Comment (outcatcher) [inline delatore/sources/awx_api.py]: let's name it `template_data`

File: delatore/sources/awx_api.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    def __init__(self):
+        self.session = requests.session()
+        self.session.headers.update({"Authorization": f"Bearer {os.getenv('AWX_AUTH_TOKEN')}"})
+        self.url = 'https://awx.eco.tsi-dev.otc-service.com/api/v2'
+
+    def create_status_message(self, template: str = None):
+        """Get last job statuses for concrete template or all templates
+
+        :param template: name of template, if empty — all scenarios
+        :param count: count of last statuses
```
Comment (outcatcher) [inline delatore/sources/awx_api.py]: No such argument

File: delatore/sources/awx_api.py [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+LOGGER.setLevel(logging.INFO)
+
+
+class TemplateStatus(NamedTuple):
+    name: str
+    last_run_timestamp: str
+    last_status: str
+    playbook: str
+
+    def __str__(self):
```
Comment (outcatcher) [inline delatore/sources/awx_api.py]: Maybe use `convert` from `json2mdwn`?

File: tests/test_awx_status.py [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+def awx_client():
+    client = AwxApiClient()
+    return client
+
+
+TEMPLATE_NAME = 'Scenario 1.5'
+
+
+def test_get_data(awx_client):
+    message = awx_client.create_status_message(template = TEMPLATE_NAME)
```
Comment (outcatcher) [inline tests/test_awx_status.py]: Redundant spaces in `(template = TEMPLATE_NAME)`

File: delatore/bot.py [hunk 8]
```diff
@@ -44,13 +47,15 @@ def __read_config():
 MARKDOWN = 'Markdown'
 
 
+
+
```
Comment (anton-sidelnikov) [inline delatore/bot.py]: Empty strings here for some reason?

**Comment by anton-sidelnikov** [comment]
Please add test for sending like test_influx_message_sending and test for check that we will send properly formatted message to chat, like  test_influx_source

File: delatore/bot.py [hunk 9]
```diff
@@ -75,3 +75,22 @@ def alert(self, text, chat):
...
+        # msg_text = "/status awx:scenario1 5"
+        message_text: str = message.text
+        _, *arguments = message_text.split(" ")
+        if not arguments:
+            return
+        if len(arguments) > 1:
+            count = int(arguments[1])
+        else:
+            count = 1
+        target: str = arguments.pop(0).split(":")
```
Comment (outcatcher) [inline delatore/bot.py]: Split return is list

File: tests/test_json2md.py [hunk 10]
```diff
@@ -1,3 +1,5 @@
+import pytest
```
Comment (outcatcher) [inline tests/test_json2md.py]: Redundant import

**Pull request merged**